emma-language/src/main/scala/org/emmalanguage/compiler/lang/source/Source.scala

```diff
@@ -157,6 +159,101 @@ trait Source extends Common
       //@formatter:on
     }
 
+    trait Algebra[A] {
+
+      // Empty tree
+      def empty: A
+
+      // Atomics
+      def lit(value: Any): A
+      def this_(encl: u.Symbol): A
+      def ref(target: u.TermSymbol): A
+
+      // References (with defaults)
+      def moduleRef(target: u.ModuleSymbol): A = ref(target)
+      def bindingRef(target: u.TermSymbol): A = ref(target)
+      def valRef(target: u.TermSymbol): A = bindingRef(target)
+      def varRef(target: u.TermSymbol): A = bindingRef(target)
+      def parRef(target: u.TermSymbol): A = bindingRef(target)
+
+      // Definitions
```

Comment:
I'm missing `defDef` but I guess it's not allowed in your source language? I had it in there to allow function definitions be translated to definitions in DML. Would a better alternative be to use the `emma.lib` annoations?